File: 1-two-sum/two-sum.go
package main

import "fmt"

type Solution struct {
}

func (*Solution) calculate(nums []int, target int) []int {
	nmap := make(map[int]int)
	for i, num := range nums {
		complement := target - num
		if j, found := nmap[complement]; found {
			return []int{i, j}
		}
		nmap[num] = i
	}
	return []int{}

}

func main() {
	solution := new(Solution)
	nums := []int{2, 7, 11, 15}
	target := 9
	result := solution.calculate(nums, target)
	fmt.Println(result)
}

File: 1-two-sum/two-sum.py
from typing import List


class Problem:
    def calculate(self, nums: List[int], target: int) -> List[int]:
        nmap: dict[int, int] = {}
        for i, num in enumerate(nums):
            complement = target - num
            if complement in nmap:
                return [i, nmap[complement]]
            nmap[num] = i
        return []

if __name__ == '__main__':
    problem = Problem()

    nums = [2,7,11,15]
    target = 9

    result = problem.calculate(nums, target)
    print(result)